BIP: Deterministic multi-signature P2SH addresses

afk11 opened the pull request:
BIP for standard multisignature P2SH addresses given m and a set of public keys.

Bip90 is just a number which doesn't collide, I will request one via the mailing list now.

btcdrak commented on bip-0090.mediawiki:
```diff
@@ -0,0 +1,119 @@
+    OP_2 021f2f6e1e50cb6a953935c3601284925decd3fd21bc445712576873fb8c6ebc18 022df8750480ad5b26950b25c7ba79d3e37d75f640f8e5d9bcd5b150a0f85014da 03e3818b65bcc73a7d64064106a859cc1a5a728c4345ff0b641209fba0d90de6e9 OP_3 OP_CHECKSIG
+
+Hash the redeem script according to BIP-0016 to get the P2SH address.
+    3Q4sF6tv9wsdqu2NtARzNCpQgwifm2rAba
```
Think you need a line here, this address isnt being displated in the rendered mediawiki view.

afk11 replied:
Thanks, seems to have been the same for the first set of public keys too. Fixed now

maraoz commented:
ACK. Can confirm bitcore follows this, and I think it's good to have a BIP to clarify this issue, given every developer doing P2SH multisig will encounter it.
Once/if this gets merged, we'll add the test vectors to bitcore :+1:

jonasschnelli commented on bip-0090.mediawiki:
```diff
@@ -0,0 +1,123 @@
+* Haskoin - https://github.com/haskoin/haskoin/blob/master/Network/Haskoin/Script/Parser.hs#L112-122
+* Armory - https://github.com/etotheipi/BitcoinArmory/blob/268db0f3fa20c989057bd43343a43b2edbe89aeb/armoryengine/ArmoryUtils.py#L1441
+* Multisignature Brainwallet - http://ms-brainwallet.org/
+    
```
I think here "bitcoinj" should also be mentioned.

rubensayshi replied:
I checked before, afaik it does not force an ordering; https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/script/ScriptBuilder.java#L168

maraoz replied:
Looks like it sorts it here: https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/script/ScriptBuilder.java#L331

rubensayshi replied:
ah, great!

afk11 commented:
Cheers @maraoz! Already borrowed the one bitcore happened to have :)

gmaxwell commented:
As a matter of procedure this proposal can not have 90 now.  _PLEASE_ discuss your proposals on the list and just use BIP?? or BIP_draft_proposer or some other placeholder name.

btcdrak commented:
@gmaxwell I thought this was discussed on the list not so long ago?

gmaxwell commented:
@btcdrak I have no freeking idea. Just immediate intervention is required to prevent ANOTHER incident of multiple people squatting a number.

afk11 commented:
That's absolutely fine, I'm not fussed either way, but can see how it can cause confusion later on.

gmaxwell commented:
Yep. Working on a last call now. Good work.

btcdrak commented:
ACK from me. I was involved in some of the discussions of this proposal in the beginning so it's all good from my side.

devrandom commented:
ACK from CryptoCorp

afk11 commented:
Whoop, we've been assigned BIP0067 for this! Updated the PR so the README reflects this now also. 

I've changed the type from Informational to Standard, seems to be what most process BIPs are marked as.

luke-jr commented on bip-0067.mediawiki:
```diff
@@ -0,0 +1,123 @@
+* Bitcore - https://github.com/bitpay/bitcore/blob/50a868cb8cdf2be04bb1c5bf4bcc064cc06f5888/lib/script/script.js#L541
+* Haskoin - https://github.com/haskoin/haskoin/blob/master/Network/Haskoin/Script/Parser.hs#L112-122
+* Armory - https://github.com/etotheipi/BitcoinArmory/blob/268db0f3fa20c989057bd43343a43b2edbe89aeb/armoryengine/ArmoryUtils.py#L1441
+* Multisignature Brainwallet - http://ms-brainwallet.org/
```
BIPs really shouldn't be listing all implementations, just a (few) reference ones)... Recommend at least not listing ones based on bad security practices like brainwallets.

rubensayshi replied:
the reason for listing as many as possible is initially too show there's already a bunch of wallets and libs following this. 

2nd is imo that this BIP aims to close the gap for compatibility, I think it's relevant. 

imo it should stay in until the BIP is accepted, since it's relevant for the discussion

afk11 commented:
@luke-jr - Forgot to mention, following your advise on IRC a while ago I altered the BIP to mention BIP-0011. Mind having a look?

rubensayshi commented:
does this mean it's accepted? or just merged the draft?

laanwj commented:
It just means that the BIP is part of the official repository now. There hasn't been a status change (those should be discussed on the mailing list, as described in BIP 0001, after the standard has been in use for significant amount of time, but it looks like the majority of BIPs stays at Draft status).
